modules/dnn/src/layers/slice_layer.cpp

```diff
@@ -209,8 +219,37 @@ class SliceLayerImpl : public SliceLayer
         CV_Assert(inputs.size() == 1);
         MatShape inpShape = inputs[0];
 
+        // in case of negative steps,
+        // x of shape [5, 10], x[5:0:-1, 10:1:-3] <=> np.flip(x[1:5:1, 2:10:3], aixs=(0, 1))
+        // new_end_i = start_i + 1 > dim_i ? dim_i : start_i + 1
+        // new_start_i = end + 1
+        // new_start_i = new_end_i - 1 - ((new_end_i - 1 - new_start_i) / abs(step_i)) * abs(step_i)
+        std::vector<std::vector<int> > sliceSteps_ = sliceSteps;
+        std::vector<std::vector<cv::Range> > sliceRanges_ = sliceRanges;
+        if (hasSteps && hasNegSteps)
```

Comment:
nit: we could invert the condition to reduce nesting

Reply:
I don't really understand how inverting the condition here can reduce nesting. Could you give me an example?